clientconn.go

```diff
@@ -118,6 +118,15 @@ func WithPicker(p Picker) DialOption {
 // WithBackoffConfig configures the dialer to use the provided backoff
 // parameters after connection failures.
 func WithBackoffConfig(b *BackoffConfig) DialOption {
```

Comment:
The current impl works but is somewhat weird. I propose the following:

WithMaxBackoffDelay(d time.Duration) DialOption {
  b := DefaultBackoffConfig
  if d > 0 {
    b.MaxDealy = d
  }
  return withbackoff(&b)
}

func WithBackoffConfig(b *BackoffConfig) DialOption {
  return WithMaxDelay(b.MaxDelay)
}